packages/jmix-react-core/src/app/Router.tsx

```diff
@@ -28,24 +29,47 @@ function exec(re: RegExp, str: string, keys: pathToRegexp.Key[]) {
   return result;
 }
 
+type FormattedTitleHandler = (caption: string) => void
+
+let formattedTitleHandler: FormattedTitleHandler | null = null
+export const registerFormattedTitleHandler = (handler: FormattedTitleHandler) => {
+  formattedTitleHandler = handler
+}
+export const clearFormattedTitleHandler = () => {
+  formattedTitleHandler = null
+}
+export const setFormattedTitle = (caption: string) => {
+  if (formattedTitleHandler == null) {
+    return
+  }
+
+  formattedTitleHandler(caption)
+}
+
```

Comment:
I suggest to simplify it / make it more direct. Code responsible for setting title shouldn't know anything about i18n, it should just update the page title with whatever string has been passed to it. If i18n is needed, it should IMO be done before calling `setTitle`.